barretenberg/cpp/src/barretenberg/benchmark/widgets_bench/widget.bench.cpp

```diff
@@ -3,15 +3,32 @@
 #include "barretenberg/flavor/ultra.hpp"
 #include "barretenberg/plonk/composer/standard_composer.hpp"
 #include "barretenberg/plonk/composer/ultra_composer.hpp"
+#include "barretenberg/plonk/proof_system/widgets/random_widgets/permutation_widget.hpp"
+#include "barretenberg/plonk/proof_system/widgets/random_widgets/plookup_widget.hpp"
+#include "barretenberg/plonk/proof_system/widgets/transition_widgets/elliptic_widget.hpp"
+#include "barretenberg/plonk/proof_system/widgets/transition_widgets/genperm_sort_widget.hpp"
+#include "barretenberg/plonk/proof_system/widgets/transition_widgets/plookup_arithmetic_widget.hpp"
 #include "barretenberg/plonk/proof_system/widgets/transition_widgets/plookup_auxiliary_widget.hpp"
-#include <benchmark/benchmark.h>
+
+// The widgets are implemented in a non-uniform way where the transition widgets provide a per-row execution function
+// `accumulate_contribution` while the random widgets do not. Defining this preprocessor variable allows to derive a
+// per-row exeuction cost that is suitable for comparing against the cost of executing the Honk relations. For
+// validation, we also directly benchmark the available `accumulate_contribution` functions.
+//
+// NOTE: this code is to be run singly threaded via taskset.
```

Comment:
```suggestion
// NOTE: this code is to be run singly threaded via taskset, e.g. taskset -c 0
```
(since I always forget the syntax for things like this)